amplfi/train/models/flow.py

```diff
@@ -164,55 +197,73 @@ def test_step(self, batch, _) -> AmplfiResult:
             descaled, mask = self.filter_parameters(descaled)
             log_probs = log_probs[mask]
 
-        parameters = self.scale(parameters, reverse=True)
-        parameters = parameters.cpu().numpy()[0]
+        # convert samples to dictionary for
+        # calculating log probabilites
+        samples_dict = dict(
+            zip(
+                self.hparams.inference_params,
+                descaled.transpose(1, 0),
+                strict=True,
+            )
+        )
 
-        # create a dictionary of injection parameters
-        # mapping from parameter string to the true injection value
-        # and add snr if provided
-        injection_parameters = {
-            k: float(v)
-            for k, v in zip(self.inference_params, parameters, strict=False)
-        }
-        injection_parameters["ra"] = injection_parameters["phi"]
+        log_priors = self.training_prior.log_prob(samples_dict)
+
+        # when predicting, there will be no ground truth parameters
+        injection_parameters = None
+        if parameters is not None:
+            parameters = self.scale(parameters, reverse=True)
+            parameters = parameters.cpu().numpy()[0]
+
+            # create a dictionary of injection parameters
+            # mapping from parameter string to the true
+            # injection value, and add snr if provided
+            injection_parameters = {
+                k: float(v)
+                for k, v in zip(
+                    self.inference_params, parameters, strict=False
+                )
+            }
+            injection_parameters["ra"] = injection_parameters["phi"]
 
-        if snr is not None:
-            injection_parameters["snr"] = snr[0].item()
+            if snr is not None:
+                injection_parameters["snr"] = snr[0].item()
 
         result = self.cast_as_bilby_result(
             descaled.cpu().numpy(),
             log_probs.cpu().numpy(),
+            log_priors.cpu().numpy(),
             injection_parameters,
         )
 
-        return result
+        # optionally reweight to different prior
+        # TODO: include likelihood reweighting
+        reweighted = None
+        if self.target_prior is not None:
+            self._logger.info(
+                f"Reweighting {len(result.posterior)} posterior samples"
+            )
+            reweighted = result.reweight_to_prior(self.target_prior)
```

Comment:
are these samples, or densities? Please clarify in variable naming.

Reply:
seems like the former, from the logging statement below. In that case, use `reweighted_samples` as the variable name.